### ISSUE TITLE
Cannot get this to work at all

### ISSUE THREAD

**Issue by mearlam** [samsface/godot-steam-api#30]
I have googled everywhere trying to find out what I am doing wrong but this plugin does not work for me in any circumstance and doesn't really say why.

- SDK downloaded, pressed the button and it said it copied the libs no problem and a green Done
- Added my game id as requested
- Tried running example scene in editor, errors
- Tried running my game from editor, errors
- Tried building and running from exe errors

- Steam is on
- steam_appid.txt is next to the exe when built 
- i put steam_appid.txt in the root directory of my res:// just in case, no difference.
- i re-installed the plugin, no difference
- I created a new project with nothing in it other than the plugin running Steam.is_init() - same issue
- **Using Godgot v3.5.stable.official [991bb6ac7]**

Any help would be appreciated, thanks.

Error from IDE:

* Is Steam running? - YES
* Is the plugin activated? - YES
*  Did you do the setup in ProjectSettings > SteamAPI? - YES

`E 0:00:09.603   call: Failed to init steam api. Is Steam running? Is the plugin activated? Did you do the setup in ProjectSettings > SteamAPI
  <C++ Source>  modules/gdscript/gdscript_functions.cpp:775 @ call()
  <Stack Trace> steam.gd:44 @ _ready()`

No other errors, this is the only one.

Error when running from exe:

* Is Steam running? - YES
* Is the plugin activated? - YES
*  Did you do the setup in ProjectSettings > SteamAPI? - YES

`
ERROR: Failed to init steam api. Is Steam running? Is the plugin activated? Did you do the setup in ProjectSettings > SteamAPI
   at: call (modules/gdscript/gdscript_functions.cpp:775) - Failed to init steam api. Is Steam running? Is the plugin activated? Did you do the setup in ProjectSettings > SteamAPI
ERROR: You're running an exported build. Steam integration will only work if the game is launched by Steam. Either upload this build to Steam or create a file in the same directory as your exported build named 'steam_appid.txt' with just your steam app id in it.
   at: call (modules/gdscript/gdscript_functions.cpp:775) - You're running an exported build. Steam integration will only work if the game is launched by Steam. Either upload this build to Steam or create a file in the same directory as your exported build named 'steam_appid.txt' with just your steam app id in it.`

**Comment by samsface** [comment]
@mearlam 

- What platform are you on?
- Did you get the plugin from the asset library or here?
- Are there anymore errors printed to the console (not the godot console, the cmd line terminal console)? You shouldn't need to export the project, it will run fine from the editor.

**Comment by mearlam** [comment]
* Windows 11
* From the asset lib, the tab in the Editor, downloaded and installed,
* I don't run it from the command line but tried it there, nothing of use. Some errors that I have caused but nothing about steam

`Running: E:/Godot_v3.5-stable_win64.exe --path E:/work/eow-survivors-go --remote-debug 127.0.0.1:6008 --allow_focus_steal_pid 131352 --position 760,180 --breakpoints res://CarTrigger.gd:9,res://UI/HoverLabelButtonUI.gd:61,res://UI/ItemUI.gd:58

Godot Engine v3.5.stable.official.991bb6ac7 - https://godotengine.org
OpenGL ES 3.0 Renderer: NVIDIA GeForce RTX 3080/PCIe/SSE2
Async. shader compilation: OFF

ERROR: Error calling method from signal 'visibility_changed': 'CanvasLayer(PauseMenuUI.gd)::_on_HomeBaseUI_visibility_changed': Method not found..
   at: emit_signal (core/object.cpp:1242)
ERROR: Failed to init steam api. Is Steam running? Is the plugin activated? Did you do the setup in ProjectSettings > SteamAPI
   at: call (modules/gdscript/gdscript_functions.cpp:775)
Steam.is_init: False
Steam achievement test: Null
ERROR: Error calling method from signal 'sort_children': 'CanvasLayer(PauseMenuUI.gd)::_on_SettingsMenu_sort_children': Method not found..`

**Comment by samsface** [comment]
Just started a fresh project on Win11, Godot Stable 3.5, downloaded from the plugin asset lib and worked straight away :thinking:.

**Comment by mearlam** [comment]
Ah hang on, I just noticed this label. 

**Steam App ID (must be a valid in your steam library).**

I think this is the problem. I only have a store page at the moment. 

Is it even possible to get my game in my steam library before it is released?

**Comment by samsface** [comment]
Ah, there's your problem. The steam game needs to be in your library.

You can request keys for your game here before it's released. Login to steamworks > landing page of your app > Request Product Keys > Developer Grant (should get you an instant key, might have to relogin to steam).

**Comment by mearlam** [comment]
So it was me doing something dumb of course. 

I have requested a key and got this lovely reply.

`Key requests are usually reviewed within a day or two. You'll receive an e-mail once your keys have been approved.`

You are correct though, I am logged in with my personal steam account but use a new one for my game that I have never logged in as.

I swapped accounts while I wait for that key and it works perfectly. 

Clearly this is me being a total clown but could you make a change in the code to give a better error about not finding the game in your library or something?

Not like I should be suggesting anything mind.

Thanks for your help and being so quick to respond.

**Comment by samsface** [comment]
Yeah there should be an error for this. Leaving ticket open to track.
